frontend/src/components/SignUp.jsx

```diff
@@ -0,0 +1,84 @@
+
+    return (
+        <>
+            <form onSubmit={() => { preventDefault() }} className='border-2 border-black flex flex-col text-center'>
+                <label htmlFor="userEmail">Email</label>
+                <div className='bg-white w-11/12 sm:w-96 border-2 rounded-full self-center flex border-black focus-within:border-blue-600 focus-within:text-blue-600'>
+                    <FontAwesomeIcon icon={faUser} className=' border-1 mt-1 ml-1 bg-none focus-input:text-blue-600'></FontAwesomeIcon>
+                    <input
+                        required
+                        type="text"
+                        id="userEmail"
+                        value={userEmail}
+                        name="userEmail"
+                        onChange={(a) => { setEmail(a.target.value); validateEmail(a, validator, setMessage) }}
+                        className='border-0 outline-none pl-1 border-black self-center flex-grow rounded-full focus:text-black' />
+                </div>
+                <span>{message}</span>
+                <label htmlFor='userPassword'>Password</label>
+                <div className='bg-white w-11/12 sm:w-96 border-2 rounded-full self-center flex border-black focus-within:border-blue-600 focus-within:text-blue-600'>
+                    <FontAwesomeIcon icon={faLock} className=' border-1 mt-1 ml-1 bg-none focus-input:text-blue-600'></FontAwesomeIcon>
+                    <input
+                        type={showPass}
+                        id="userPassword"
+                        name="userPassword"
+                        value={userPassword}
+                        onChange={(a) => { setPassword(a.target.value) }}
+                        className='border-0 outline-none pl-1 border-black self-center flex-grow rounded-full focus:text-black' />
+                </div>
+                <label htmlFor='confirmPass'>ReType-Password</label>
+                <div className='bg-white w-11/12 sm:w-96 border-2 rounded-full self-center flex border-black focus-within:border-blue-600 focus-within:text-blue-600'>
+                    <FontAwesomeIcon icon={faLock} className=' border-1 mt-1 ml-1 bg-none focus-input:text-blue-600'></FontAwesomeIcon>
+                    <input
+                        type={showPass}
+                        id="confirmPass"
+                        name="confirmPass"
+                        value={confirmPass}
+                        onChange={(a) => { setConPass(a.target.value) }}
+                        className='border-0 outline-none pl-1 border-black self-center flex-grow rounded-full focus:text-black' />
+                </div>
+                <span className='self-center text-left w-11/12 sm:w-96 border-none'>
+                    <input type='checkBox' className='checkBox' id='showPass' onClick={(e) => { showPassword(e, setPass) }}></input><label htmlFor='showPass'>Show Password</label>
+                </span>
+                <input type='submit' onClick={(e) => {
```

Comment:
don't add functions directly into html, if function is more than of two lines then create a function at top level.

Reply:
Done